zipline/pipeline/term.py

```diff
@@ -500,6 +525,64 @@ def __repr__(self):
         )
 
 
+class Slice(ComputableTerm):
+    """
+    Term for extracting a single column of a another term's output.
+
+    Parameters
+    ----------
+    term : zipline.pipeline.term.Term
+        The term from which to extract a column of data.
+    asset : zipline.assets.Asset
+        The asset corresponding to the column of `term` to be extracted.
+
+    Notes
+    -----
+    Users should rarely construct instances of `Slice` directly. Instead, they
+    should construct instances via indexing, e.g. `MyFactor()[Asset(24)]`.
+    """
+    def __new__(cls, term, asset):
+        return super(Slice, cls).__new__(
+            cls,
+            asset=asset,
+            inputs=[term],
+            window_length=0,
+            mask=term.mask[asset] if term.mask else AssetExists(),
```

Comment:
Since we are taking the slice of our term's mask, this triggers a recursive call to `__new__`, so the `else` is the base case of when `term` is `AssetExists` and `term.mask` is `None`. Is this the correct way to handle this?